frontend/src/metabase/visualizations/components/ChartSettingsWidgetPopover.tsx

```diff
@@ -13,22 +13,55 @@ interface Widget {
 interface ChartSettingsWidgetPopoverProps {
   anchor: HTMLElement;
   handleEndShowWidget: () => void;
-  widget: Widget;
+  widgets: Widget[];
+  currentWidgetKey: string;
 }
 
 const ChartSettingsWidgetPopover = ({
   anchor,
   handleEndShowWidget,
-  widget,
+  widgets,
+  currentWidgetKey,
 }: ChartSettingsWidgetPopoverProps) => {
+  const sections = useMemo(() => {
+    return _.groupBy(widgets, "section");
+  }, [widgets]);
+
+  const [currentSection, setCurrentSection] = useState<React.Key>("");
+  const [widgetKey, setWidgetKey] = useState(currentWidgetKey);
+
+  useEffect(() => {
+    if (currentWidgetKey !== widgetKey) {
+      setCurrentSection(Object.keys(sections)[0]);
+      setWidgetKey(currentWidgetKey);
```

Comment:
Now I understand what you were trying to do now. TBH your first approach looks easier to understand haha. But I think using our `usePrevious` hook would make this part express itself better. So it could look something like this
```ts
  const sections = useMemo(() => {
    return _.groupBy(widgets, "section");
  }, [widgets]);

  const [currentSection, setCurrentSection] = useState<React.Key>("");
  const previousWidgetKey = usePrevious(currentWidgetKey);

  useEffect(() => {
    if (currentWidgetKey !== previousWidgetKey) {
      setCurrentSection(Object.keys(sections)[0]);
    }
  }, [currentWidgetKey, previousWidgetKey, sections]);
```

I have also verified locally that it still works when you change values on the second tab e.g. style.

Reply:
I didn't know we had such a hook! I agree that this is probably easier to understand. I'll update it 👍